src/pushNotifications/TokenHandling.ts

```diff
@@ -0,0 +1,75 @@
+import { Platform } from 'react-native';
+
+import { secrets } from '../config';
+import * as SecureStore from 'expo-secure-store';
+import * as appJson from '../../app.json';
+
+
+const namespace = appJson.expo.slug as keyof typeof secrets;
+
+export enum PushNotificationStorageKeys {
+    PUSH_TOKEN = 'PUSH_TOKEN',
+    IN_APP_PERMISSION = 'IN_APP_PERMISSION',
+}
+
+export const handleIncomingToken = async (token?: string) => {
+  // eslint-disable-next-line no-console
+  console.log(token); // FIXME remove for production
+
+  await getTokenFromStorage().then((result) => {
+    if (result != (token ?? null)) {
+      if (result) removeTokenFromServer(result);
+      if (token) addTokenToServer(token);
+      storeTokenSecurely(token);
+    }
+  });
+};
+
+const removeTokenFromServer = async (token: string) => {
+  const accessToken = await SecureStore.getItemAsync('ACCESS_TOKEN');
+  const requestPath = secrets[namespace].serverUrl + secrets[namespace].rest.pushDevicesDelete;
+  const fetchObj = {
+    method: 'DELETE',
+    headers: {
+      Accept: 'application/json',
+      Authorization: 'Bearer ' + accessToken,
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify({
+      token
+    })
+  };
+
+  if (accessToken) fetch(requestPath, fetchObj);
+};
+
+const addTokenToServer = async (token: string) => {
+  const accessToken = await SecureStore.getItemAsync('ACCESS_TOKEN');
+  const requestPath = secrets[namespace].serverUrl + secrets[namespace].rest.pushDevicesRegister;
+  const os = Platform.OS === 'ios' || Platform.OS === 'android' ? Platform.OS : 'undefined';
```

Comment:
isn't a `Platform.OS` enough here? maybe `toString()` to have the correct type even if it is `undefined`?
`device.platform` should be used here, as pointed out [here](https://github.com/ikuseiGmbH/smart-village-app-app/pull/124/files#r529742692).

other possibility would be:
```
Platform.select({
  android: 'android',
  ios: 'ios',
  default: 'undefined'
})
```

Reply:
Using `Platform.select` changes the type that can be inferred by ts from this
<img width="365" alt="Screenshot 2020-11-25 at 09 09 59" src="https://user-images.githubusercontent.com/59824597/100199802-01b2d480-2efe-11eb-8ab3-c3e480f24ebb.png">
to this
<img width="151" alt="Screenshot 2020-11-25 at 09 10 29" src="https://user-images.githubusercontent.com/59824597/100199850-12634a80-2efe-11eb-9716-ea7bf0a01f7d.png">

This is not a big issue, due to the fact that we just need it to be a `string` to send it in the json though.

The reason this is mapped to just those three values is, because our the backend implementation handles these three cases, and not `"macos" | "windows" | "web"`.

I will change it to use `device.platform` instead of `Platform.OS`, but leave the structure as is.